dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -10,346 +13,208 @@
-	private String h2Create = "CREATE TABLE db_version (db_version integer NOT NULL, date_update timestamp NOT NULL, CONSTRAINT db_version_pkey PRIMARY KEY (db_version))";
-	
-	private String pgSelect = "SELECT max(db_version) AS db_version FROM db_version";
-	private String mySelect = "SELECT max(db_version) AS db_version FROM db_version";
-	private String oraSelect = "SELECT max(db_version) AS db_version FROM db_version";
-	private String msSelect = "SELECT max(db_version) AS db_version FROM db_version";
-	private String h2Select = "SELECT max(db_version) AS db_version FROM db_version";
 
-	private String lock;
-	private String commit;
-	private String create;
-	private String select;
 
 
+	private final String pgCreate  = "CREATE TABLE db_version (db_version integer NOT NULL, date_update timestamp with time zone NOT NULL, CONSTRAINT db_version_pkey PRIMARY KEY (db_version));";
+	private final String myCreate  = "CREATE TABLE `db_version` (`db_version` INTEGER UNSIGNED NOT NULL,`date_update` DATETIME NOT NULL, PRIMARY KEY (`db_version`))";
+	private final String oraCreate = "CREATE TABLE \"DB_VERSION\" ( \"DB_VERSION\" INTEGER NOT NULL , \"DATE_UPDATE\" TIMESTAMP NOT NULL, PRIMARY KEY (\"DB_VERSION\") )";
+	private final String msCreate  = "CREATE TABLE db_version (	db_version int NOT NULL , date_update datetime NOT NULL, PRIMARY KEY (db_version) )";
+
+	private final String select = "SELECT count(*) as count_versions, max(db_version) AS db_version FROM db_version";
+
+
+
+	final boolean firstTimeStart;
 	
 	
 	private StartupTasksExecutor() {
 
+	    insureDbVersionTable();
+	    Config.DB_VERSION = currentDbVersion();
+        this.firstTimeStart = (Config.DB_VERSION==0);
+	    
 	}
 
-	public static StartupTasksExecutor getInstance() {
+	public static synchronized StartupTasksExecutor getInstance() {
 		if (executor == null)
 			executor = new StartupTasksExecutor();
 		return executor;
 	}
 
-	/**
-	 * Check which database we're using, and select the apropiate SQL. In a
-	 * different method to avoid further clutter
-	 */
-	private void setupSQL() {
-		if (DbConnectionFactory.isPostgres()) {
-			lock = pgLock;
-			commit = pgCommit;
-			create = pgCreate;
-			select = pgSelect;
-		}
-		else if (DbConnectionFactory.isMySql()) {
-			lock = myLock.toLowerCase();
-			commit = myCommit.toLowerCase();
-			create = myCreate.toLowerCase();
-			select = mySelect.toLowerCase();
-		}
-		
-		else if (DbConnectionFactory.isOracle()) {
-			lock = oraLock;
-			commit = oraCommit;
-			create = oraCreate;
-			select = oraSelect;
-		}
-		
-		else if (DbConnectionFactory.isMsSql()) {
-			lock = msLock;
-			commit = msCommit;
-			create = msCreate;
-			select = msSelect;
-		}
-		
-		else if(DbConnectionFactory.isH2()) {
-		    lock = h2Lock;
-		    commit = h2Commit;
-		    create = h2Create;
-		    select = h2Select;
-		}
 
+	private final String createTableSQL() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'final' on method 'createTableSQL': private methods cannot be overridden](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)